src/components/Pages/Submit/Form/Tags/index.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { FC, useCallback, useEffect, useRef, useState } from "react";
+import { Tag, TagGroup, Input } from "rsuite";
+import { ArticleParams } from "../store/types";
+import { AddLink } from "../../../../Icons";
+
+const removeTag = (
+  tag: string,
+  tags: string[],
+  upsert: (update: ArticleParams) => void
+) => {
+  const nextTags = tags.filter((item) => item !== tag);
+  upsert({ tags: nextTags });
+};
+
+const Tags: FC<{
+  tags: string[];
+  upsert: (update: ArticleParams) => void;
+}> = ({ tags, upsert }) => {
+  const [isEditing, setEditing] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const update = useCallback(
+    (tag: string) => {
+      if (tag == "") {
+        setEditing(false);
+        return;
+      }
+      upsert({ tags: [...tags, tag] });
+    },
+    [tags, setEditing]
+  );
+
+  useEffect(() => {
+    if (isEditing && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isEditing]);
+
+  useEffect(() => {
+    setEditing(false);
+  }, [tags]);
+
+  return (
+    <TagGroup>
+      {tags.map((item, index) => (
+        <Tag key={index} closable onClose={() => removeTag(item, tags, upsert)}>
+          {item}
+        </Tag>
+      ))}
+      {!isEditing && (
+        <a className="btn" onClick={() => setEditing(true)}>
+          <AddLink />
+        </a>
+      )}
+      {isEditing && (
```

Comment:
I didn't see you use this in your code, in JSX you can use a boolean to run or not run a function call, so here isEditing is controlling if the a link or the input is rendered. You can chain booleans for more complex cases